finesse/hardware/em27_scraper.py

```diff
@@ -0,0 +1,128 @@
+"""This module provides an interface to the EM27 monitor.
+
+This is used to scrape the PSF27 sensor data table off the server.
+"""
+import logging
+from dataclasses import dataclass
+from decimal import Decimal
+
+from pubsub import pub
+from requests import get
+
+from ..config import EM27_URL
+
+
+@dataclass
+class EM27Property:
+    """Class for representing EM27 monitored properties.
+
+    Args:
+        name: name of the physical quantity
+        value: value of the physical quantity
+        unit: unit in which the value is presented
+    """
+
+    name: str
+    value: Decimal
+    unit: str
+
+    def __str__(self) -> str:
+        """Print a property's name, value and unit in a readable format.
+
+        Returns:
+            str: The name, value and unit of a property.
+        """
+        return f"{self.name} = {self.value:.6f} {self.unit}"
+
+    def val_str(self) -> str:
+        """Print a property's value and unit in required format.
+
+        Returns:
+            str: The value and unit of a property in the format consistent with
+                 the previous FINESSE GUI.
+        """
+        return f"{self.value:.6f} {self.unit}"
+
+
+class PSF27Error(Exception):
```

Comment:
Honestly, no idea. Maybe point spread function if it's monitoring the optics? The table on the webpage is titled "EM27 SN_93 Automation units Diagnostics" which contains a load of information, some of which comes under sections whose titles start with "PSF27", so it seems like PSF27 is just a specific part of the EM27. The specific table I search for comes under "PSF27Sensor" so that's what I went with. Even more confusing is that the GUI panel which shows these is titled "EM27 SOH Monitor". Maybe I should ask Jon.